ddtrace/tracer/abandoned_spans_test.go

```diff
@@ -0,0 +1,219 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var warnPrefix = fmt.Sprintf("Datadog Tracer %v WARN: ", version.Tag)
+var spanStart = time.Date(2023, time.August, 18, 0, 0, 0, 0, time.UTC)
+
+// setTestTime() sets the current time, which will be used to calculate the
+// duration of abandoned spans.
+func setTestTime() {
+	current := spanStart.UnixNano() + 10*time.Minute.Nanoseconds() //use a fixed time instead of now
+	now = func() int64 { return current }
+}
+
+// resetTestTime() resets the `now()` function to its default value.
+func resetTestTime() {
+	now = func() int64 { return time.Now().UnixNano() }
+}
+
+func spanAge(s *span) string {
+	return fmt.Sprintf("%d sec", (now()-s.Start)/1e9)
```

Comment:
instead of using 1e9 here it might be a bit more readable to just use `time.Second` (Although the different type might actually make it worse so feel free to ignore this if it's grosser)